### DISCUSSION TITLE
Backup and read NoiseCapture local Android database

### DISCUSSION THREAD

**Discussion by nicolas-f** [Ifsttar/NoiseCapture#397] [General]
You can backup the database but it requires some knowledge in Android.

The data of NoiseCapture is stored in a private folder in your phone:

/data/data/org.noise_planet.noisecapture/databases/Storage.db

You can use the device explorer of Android Studio to retrieve this file. It is a SQLLite database without passwords, so you can open it with DBeaver for example.

**Comment by yudlab** [comment]
Hello guys, having a way to access the database directly would enable many users to have more flexibility to work with the data. I tried to locate the database in /data/data/org.noise_planet.noisecapture but couldn't find it.
I'll really appreciate if someone could help me there.
I used adb over USB without success.

UPDATE ON THIS:
It appears that the database is saved in /data/user/0/org.noise_planet.noisecapture.
However there is no way to access this folder as Android(On V13 & V14) restricts access to it. (Tried over ADB)

Unfortunately since this app is not debuggable, we can't run the command below to confirm if it would be accessible this way:
`Command: user:/data/user/0 $ run-as org.noise_planet.noisecapture` returns
> run-as: package not debuggable: org.noise_planet.noisecapture

Using this website: https://apkeditor.io/ I was able to enable the debug flag in the manifest for the app, and I could successfully pull the Storage.db file using the below command:
`adb exec-out run-as org.noise_planet.noisecapture cat /data/user/0/org.noise_planet.noisecapture/databases/Storage.db > test.db`
